search/search_service/proxy/es_search_proxy.py

```diff
@@ -319,3 +326,149 @@ def search(self, *,
                                                    resource_types=resource_types)
 
         return formatted_response
+
+    def get_document_id_and_field_value_from_key(self,
+                                                 resource_key: str,
+                                                 resource_type: Resource,
+                                                 field: str) -> Tuple[str, Any]:
+        key_query = {
+            resource_type: Q(TERM_QUERY, key=resource_key),
+        }
+        response: Response = self.execute_queries(queries=key_query,
+                                                  page_index=0,
+                                                  results_per_page=1)
+        if len(response) < 1:
+            msg = f'No response from ES for key query {key_query[resource_type]}'
+            LOGGER.error(msg)
+            raise ElasticsearchException(msg)
+        if len(response) > 1:
+            msg = f'Key {key_query[resource_type]} is not unique to a single ES resource'
+            LOGGER.error(msg)
+            raise ElasticsearchException(msg)
+
+        response = response[0]
+        if response.success():
+            results_count = response.hits.total.value
+            if results_count > 0:
```

Comment:
wait weren't you supposed to do that here

Reply:
oh my bad i got confused, checking the number o responses from m search first, then checking the number of results in the particular response, I'll fix it